sdk/include/opentelemetry/sdk/logs/logger_provider.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "opentelemetry/logs/logger_provider.h"
+#include "opentelemetry/logs/noop.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/sdk/logs/logger.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+// Define the maximum number of loggers that are allowed to be registered to the loggerprovider.
+// TODO: Add link to logging spec once this is added to it
+#define MAX_LOGGER_COUNT 100
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class LoggerProvider final : public opentelemetry::logs::LoggerProvider
+{
+public:
+  /**
+   * Initialize a new logger provider with a specified processor
+   * @param processor The processor for this logger provider. This must
+   * not be a nullptr.
+   */
+  explicit LoggerProvider(std::shared_ptr<LogProcessor> processor) noexcept;
+
+  /**
+   * Initialize a new logger provider. A processor must later be assigned
+   * to this logger provider via the SetProcessor() method.
+   */
+  explicit LoggerProvider() noexcept;
+
+  /**
+   * Creates a logger with the given name, and returns a shared pointer to it.
+   * If a logger with that name already exists, return a shared pointer to it
+   * @param name The name of the logger to be created.
+   * @param options (OPTIONAL) The options for the logger. TODO: Give list of options
+   */
+  opentelemetry::nostd::shared_ptr<opentelemetry::logs::Logger> GetLogger(
```

Comment:
Could `GetOrCreateLogger` be a better name? Get implies it is very lightweight but actually it could create if the given named logger doesn't exist. As this is just `override`, we may need to change the API if take this.

Reply:
Yea GetLogger() is a bit misleading since doesn't only get loggers but can also create new loggers. Log4cxx and Log4j do it this way though so I figure we should stay as close to their implementation as possible.

Reply:
I think GetLogger is a good name, as it is in alignment with all popular log4* logging frameworks. I like this name.

Reply:
Also matches slf4j.    I agree it's a misleading BUT industry standard name.